packages/webpack-cli/lib/groups/ZeroConfigGroup.js

```diff
@@ -21,10 +19,6 @@ class ZeroConfigGroup extends GroupHelper {
             return process.env.NODE_ENV;
         } else {
             if (this.args.mode) {
-                if (this.args.mode !== PRODUCTION && this.args.mode !== DEVELOPMENT && this.args.mode !== NONE) {
-                    logger.warn('You provided an invalid value for "mode" option. Using "production" by default');
-                    return PRODUCTION;
-                }
                 return this.args.mode;
             }
             return PRODUCTION;
```

Comment:
We even simplify - just set mode only when it passed, webpack itself set `production` by default